pilot/pkg/config/memory/monitor.go

```diff
@@ -72,32 +79,45 @@ func newBufferedMonitor(store model.ConfigStore, bufferSize int, sync bool) Moni
 		store:    store,
 		handlers: handlers,
 		eventCh:  make(chan ConfigEvent, bufferSize),
-		sync:     sync,
+		sync:     syncMode,
+		lock:     &sync.RWMutex{},
 	}
 }
 
 func (m *configStoreMonitor) ScheduleProcessEvent(configEvent ConfigEvent) {
+	m.lock.Lock()
+	if m.closed.Load() {
+		m.lock.Unlock()
+		return
+	}
+
 	if m.sync {
+		m.lock.Unlock()
```

Comment:
why need to protect sync mode? It seems unneccessary

Reply:
The code is confused here, i tweak it.